cosmovisor/process.go

```diff
@@ -153,7 +154,10 @@ func doBackup(cfg *Config) error {
 }
 
 // doPreUpgrade runs the pre-upgrade command defined by the application
-func doPreUpgrade(cfg *Config) error {
+// cfg contains the cosmovisor config from env var
+// counter is the current attempt at preupgrade by cosmovisor.
+func doPreUpgrade(cfg *Config, counter int) error {
```

Comment:
Was there any particular reason to make this recursive instead of just iterating in a loop (0...cfg.PreupgradeMaxRetries)?

Reply:
@alexanderbez Recursive seemed easier to understand the logical flow. Would you recommend the iterative approach?

Reply:
I feel as if recursive is always _harder_ to understand. I would prefer to make this iterative, but it's not a blocker.